Content.Server/Chat/Systems/ChatSystem.cs

```diff
@@ -743,6 +754,15 @@ private bool CanSendInGame(string message, IConsoleShell? shell = null, ICommonS
         return !_chatManager.MessageCharacterLimit(player, message);
     }
 
+    public void HandleLastMessageBeforeDeath(EntityUid source, ICommonSession player, string message)
+    {
+
+        if (player != null && source != null)
+        {
+            _lastMessageBeforeDeathSystem.AddMessage(source, player, message, Identity.Name(source, EntityManager));
```

Comment:
`Identity.Name` still just reports their perceived name, not their true name. This still isn't correct.